File: backend/src/chat/dto/create-chat.dto.ts
import { IsEnum, IsNotEmpty, IsOptional, IsString } from 'class-validator';

export class CreateChatDto {
    @IsString()
    @IsNotEmpty()
    adId: string;
  
    @IsString()
    @IsNotEmpty()
    userId: string;

    @IsString()
    @IsNotEmpty()
    userEmail: string;
  
    @IsString()
    @IsNotEmpty()
    message: string;
  
    @IsNotEmpty()
    timestamp: Date;
  
}

File: backend-dotnet/Soa.Contracts/Dtos/CreateChatDto.cs
using System.Text.Json.Serialization;

namespace Soa.Contracts.Dtos;

public class CreateChatDto
{
    [JsonPropertyName("adId")]
    public string AdId { get; set; } = null!;

    [JsonPropertyName("userId")]
    public string UserId { get; set; } = null!;

    [JsonPropertyName("userEmail")]
    public string UserEmail { get; set; } = null!;

    [JsonPropertyName("message")]
    public string Message { get; set; } = null!;

    [JsonPropertyName("timestamp")]
    public DateTime Timestamp { get; set; }
}
